src/meson.build

```diff
@@ -4,13 +4,15 @@ numeric_sources = [
     'numeric-types.c',
     'numeric-vector.c']
 numeric_glib_lib = library('numeric-glib-1.0', numeric_sources,
-                           dependencies: [glib, gobject],
+                           dependencies: [glib, gobject, dfp],
                            c_args: '-DNUMERIC_COMPILATION',
                            install: true)
 
-numeric_glib = declare_dependency(link_with: numeric_glib_lib,
-                                  include_directories: include_directories('.'),
-                                  sources: 'numeric-glib-1.0.vapi')
+numeric_glib_dep = declare_dependency(link_with: numeric_glib_lib,
+                                    include_directories: include_directories('.'))
+
+numeric_glib_vala_dep = declare_dependency(dependencies: numeric_glib_dep,
+                                           sources: 'numeric-glib-1.0.vapi')
```

Comment:
The `sources` tag breaks C executable because Meson thinks they are Vala targets. This is why we're splitting the dependency in two pieces.